server/src/main/java/org/elasticsearch/rest/action/document/RestBulkAction.java

```diff
@@ -135,6 +139,10 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         }
     }
 
+    private static ElasticsearchParseException parseFailureException(Exception e) {
+        return new ElasticsearchParseException("could not parse bulk request body", e);
```

Comment:
Can we special case `IllegalArgumentException` to avoid the extra wrapping of it?

Reply:
Yes. I made this change.